### PR TITLE
ESQL: Add known issue to 8.15 docs for OOM due to wide index pattern

### PR DESCRIPTION
**Pull request by alex-spies**

### CODE REVIEW

**Comment by alex-spies** [comment]
Hey @esdocs , could you have a look and tell me if I'm doing this correctly?

I'm trying to add known issues to the already released versions 8.15.0 and 8.15.1.

I couldn't find release notes for 8.15.1 on main, so I went and added them - is there another place or branch that this should go into instead?

File: docs/reference/release-notes/8.15.0.asciidoc [hunk 1]
```diff
@@ -30,6 +30,12 @@ signed integer) may encounter errors (issue: {es-issue}111854[#111854])
 `xpack.security.authc.realms.*.files.role_mapping` configuration option. As a workaround, custom role mappings
 can be configured using the https://www.elastic.co/guide/en/elasticsearch/reference/current/security-api-put-role-mapping.html[REST API] (issue: {es-issue}112503[#112503])
 
+* ES|QL queries can cause nodes to crash due to Out Of Memory errors under some circumstances when there are many indices matching the pattern,
+many conflicting field mappings in those indices, and many of those fields are kept in the request.Lower memory proportional to all of that
```
Comment (nik9000) [inline docs/reference/release-notes/8.15.0.asciidoc:34]: s/request.Lower/request. Lower/

Maybe s/memory/heap/

File: docs/reference/release-notes/8.15.0.asciidoc [hunk 2]
```diff
@@ -30,6 +30,12 @@ signed integer) may encounter errors (issue: {es-issue}111854[#111854])
...
+makes it worse. (issue: {es-issue}111964[#111964], {es-issue}111358[#111358]).
+Either downgrade to 8.14, upgrade to 8.15.2 upon release, or else follow the recommendation in the manual to
```
Comment (nik9000) [inline docs/reference/release-notes/8.15.0.asciidoc:36]: Downgrade is quite difficult in ES.

**Comment by alex-spies** [comment]
I think this needs to go in the 8.15 branch, `main` doesn't have the 8.15.1 release notes. Closing this in favor of https://github.com/elastic/elasticsearch/pull/112930, which is on the 8.15 branch.

**Comment by leemthompo** [comment]
> I think this needs to go in the 8.15 branch, main doesn't have the 8.15.1 release notes. Closing this in favor of https://github.com/elastic/elasticsearch/pull/112930, which is on the 8.15 branch.

@alex-spies euggh sorry about that. I just merged the forward porting of 8.15.1 release notes https://github.com/elastic/elasticsearch/pull/112568, so really should target the `main` branch in your PR alright and backport.

**Comment by leemthompo** [comment]
I had suggested rewording in [this comment](https://github.com/elastic/elasticsearch/pull/112930#discussion_r1761332879)

**Comment by alex-spies** [comment]
@leemthompo and @tylerperk , I took over Liam's version (much nicer), added the downgrade option back in, and the suggestion to use a different default view is also in there.

Let me know if you like it this way - if yes, I think this is good to go :)

**Comment by leemthompo** [comment]
I agree with @nik9000 that downgrading is not easy, and similarly waiting for 8.15.2 isn't ideal. But they are options so I think it makes sense to have them. The others are more practical and quicker to implement.

**Review by leemthompo — APPROVED** [review]

**Review by tylerperk — APPROVED** [review]
LGTM thanks

File: docs/reference/release-notes/8.15.1.asciidoc [hunk 3]
```diff
@@ -10,6 +10,19 @@ Also see <<breaking-changes-8.15,Breaking changes in 8.15>>.
...
+** Multiple indices match the query pattern
+** These indices have many conflicting field mappings
+** Many of those fields are included in the request
+These issues deplete heap memory, increasing the likelihood of OOM errors. (issue: {es-issue}111964[#111964], {es-issue}111358[#111358]).
++
+To work around this issue, you have a number of options:
+** Downgrade to an earlier version
+** Upgrade to 8.15.2 upon release
+** Follow the instructions to
+https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-kibana.html#esql-kibana-enable)[disable ES|QL] queries in {kib}
```
Comment (leemthompo) [inline docs/reference/release-notes/8.15.1.asciidoc:23]:
```suggestion
<<esql-kibana-enable,disable ES|QL queries in {kib}>>
```

File: docs/reference/release-notes/8.15.0.asciidoc [hunk 4]
```diff
@@ -30,6 +30,19 @@ signed integer) may encounter errors (issue: {es-issue}111854[#111854])
...
+** Multiple indices match the query pattern
+** These indices have many conflicting field mappings
+** Many of those fields are included in the request
+These issues deplete heap memory, increasing the likelihood of OOM errors. (issue: {es-issue}111964[#111964], {es-issue}111358[#111358]).
++
+To work around this issue, you have a number of options:
+** Downgrade to an earlier version
+** Upgrade to 8.15.2 upon release
+** Follow the instructions to
+https://www.elastic.co/guide/en/elasticsearch/reference/current/esql-kibana.html#esql-kibana-enable)[disable ES|QL] queries in {kib}
```
Comment (leemthompo) [inline docs/reference/release-notes/8.15.0.asciidoc:43]:
```suggestion
<<esql-kibana-enable,disable ES|QL queries in {kib}>>
```

**Review by leemthompo — APPROVED** [review]

File: docs/reference/release-notes/8.15.0.asciidoc (hunk 4 above)
Comment (nik9000) [reply docs/reference/release-notes/8.15.0.asciidoc]: You beat me to it.

**Comment by nik9000** [comment]
Backported to 8.x with https://github.com/elastic/elasticsearch/pull/112959

**Pull request merged**